packages/components/src/popover/index.js

```diff
@@ -170,20 +176,33 @@ const Popover = (
 		}
 
 		const iframeRect = frameElement.getBoundingClientRect();
-		return {
-			name: 'iframeOffset',
-			fn( { x, y } ) {
-				return {
-					x: x + iframeRect.left,
-					y: y + iframeRect.top,
-				};
-			},
-		};
+		return { x: iframeRect.left, y: iframeRect.top };
 	}, [ ownerDocument ] );
 
 	const middlewares = [
-		frameOffset,
 		offset ? offsetMiddleware( offset ) : undefined,
+		frameOffset
+			? offsetMiddleware( ( { placement: currentPlacement } ) => {
+					const mainAxis = getMainAxis( currentPlacement );
+					const crossAxis = getCrossAxis( currentPlacement );
+
+					// When the popover is before the reference, subtract the offset,
+					// else add it.
+					const mainAxisModifier = currentPlacement.includes( 'top' )
+						? -1
+						: 1;
+					const crossAxisModifier = currentPlacement.includes(
+						'left'
+					)
+						? -1
+						: 1;
```

Comment:
This seems to work for top or bottom placements. Less sure if it's correct for left or right placements 🤔

Reply:
There was an issue, now resolved.